aws_protocol_adaptor/device_client/aws_nvmsgbroker.c

```diff
@@ -300,3 +355,84 @@ void nvds_msgapi_do_work(NvDsMsgApiHandle h_ptr)
 	last_send_time_stamp = current_time_stamp;
 	return;
 }
+
+char *nvds_msgapi_getversion()
+{
+	return (char *)NVDS_MSGAPI_VERSION;
+}
+
+char *nvds_msgapi_get_protocol_name()
+{
+	return (char *)NVDS_MSGAPI_PROTOCOL;
+}
+
+bool is_valid_connection_str(char *connection_str)
+{
+	char *burl = "", *bport = "";
+	if (connection_str == NULL)
+	{
+		IOT_ERROR("connection string cant be NULL");
+		return false;
+	}
+
+	char conn_str[] = connection_str;
+	int i = 0;
+
+	char *token = strtok(conn_str, ";");
+	char *data[2];
+
+	while (token)
+	{
+		data[i++] = token;
+		token = strtok(NULL, ";");
+	}
+	for (i = 0; i < 2; i++)
+	{
+		printf("%s\n", data[i]);
+	}
+	burl = data[0];
+	bport = data[1];
+
+	if (burl == "" || bport == "")
+	{
+		IOT_ERROR("connection string is invalid. hostname or port is empty\n");
+		return false;
+	}
+	return true;
+}
+
+char *generate_sha256_hash(char *str)
+{
+	unsigned char hashval[SHA256_DIGEST_LENGTH];
+	int len = SHA256_DIGEST_LENGTH * 2 + 1;
+	char res[len];
+	SHA256_CTX sha256;
+	SHA256_Init(&sha256);
+	SHA256_Update(&sha256, str, strlen(str));
+	SHA256_Final(hashval, &sha256);
+	for (int i = 0; i < SHA256_DIGEST_LENGTH; i++)
+	{
+		sprintf(res + (i * 2), "%02x", hashval[i]);
+	}
+	return res;
+}
+
+NvDsMsgApiErrorType nvds_msgapi_connection_signature(char *broker_str, char *cfg, char *output_str, int max_len)
+{
+	strcpy(output_str, "");
+	if (broker_str == NULL || cfg == NULL)
+	{
+		IOT_ERROR("nvds_msgapi_connection_signature: broker_str or cfg path cant be NULL\n");
+		return NVDS_MSGAPI_ERR;
+	}
+	if(max_len < 2 * SHA256_DIGEST_LENGTH + 1) {
```

Comment:
Can you explain?

Also, please add a space after "if"

Reply:
The Sha_digest_length is defined as 32 in the openssl/sha.h so 2*32+1= 65, so if the output_str is less than 65 chars it would return an error

Reply:
@Shubhamturakhia Consider defining a macro instead. This is hard to read.